File: perl/fibonacci.pl
#!/usr/bin/env perl
use strict;
use warnings;

use Time::HiRes qw( time );

sub fib {
    my $n = $_[0];
    return 1 if $n < 2;
    fib($n - 2) + fib($n - 1);
}

my $start = time();
my $answer = fib(40);
my $elapsed = time() - $start;

printf "%d, elapsed: %.3f secs\n", $answer, $elapsed;

File: rust/fibonacci.rs
extern crate time;

fn fib(n: i32) -> i32 {
    return if n < 2 {
        1
    } else {
        fib(n - 2) + fib(n - 1)
    }
}

fn main() {
    let start = time::pricse_time_ns();
    let answer = fib(40);
    println!("{}", answer);
}
